### PR TITLE
Feature/activity item navigate buttons

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/entities/event/model/operations/ScheduledDateCalculator.ts [hunk 1]
```diff
@@ -34,9 +34,10 @@ const calculateForMonthly = (selectedDate: Date, availability: EventAvailability
 }
 
 const calculateForSpecificDay = (specificDay: Date, availability: EventAvailability): Date | null => {
-  if (specificDay > startOfDay(new Date())) {
-    return null
-  }
+  // Ask Aleksei about it 06.03.2023, Mon
```
Comment (BamMironov) [inline src/entities/event/model/operations/ScheduledDateCalculator.ts:37]: Did you ask? :)

File: src/widgets/ActivityDetails/model/hooks/useActivityInProgress.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  const activityInProgress = activitiesInProgress.find(
+    ({ activityId, eventId }) => activityId === selectedActivityId && eventId === eventIdByActivityId,
+  )
+
+  const calculateActivityProgress = (): ActivityItem[] => {
+    let items: ActivityItem[] = []
+
+    if (isOnePageAssessment) {
+      items = activityDetails.items.map(item => item)
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/model/hooks/useActivityInProgress.ts:31]: You could just assign as is. Or if you need a copy, then [...activityDetails.items]

File: src/widgets/ActivityDetails/model/hooks/useActivityInProgress.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    } else {
+      items = activityDetails.items.reduce((acc, item, index) => {
+        if (index < activitiesInProgress.length) {
+          acc.push(item)
+        }
+
+        return acc
+      }, [] as ActivityItem[])
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/model/hooks/useActivityInProgress.ts:39]: Pass the ActivityItem type to `.reduce` function as a generic instead.

File: src/widgets/ActivityDetails/ui/ActivityItemList.tsx [hunk 4]
```diff
@@ -1,19 +1,37 @@
...
   const isOnePageAssessment = activityDetails.showAllAtOnce
   const isSummaryScreen = false // Mock
+
+  const buttonsConfig: ItemCardButtonsConfig = {
+    isOnePageAssessment,
+    isBackShown: activityDetails.items.length > 1,
+    isSubmitShown: isOnePageAssessment && activityDetails.items.length === activityInProgress?.answers.length,
+    isSkippable: activityDetails.isSkippable,
+    isNextDisable: true, // Default value === TRUE  (Condition if answer value empty or not exist)
+  }
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/ui/ActivityItemList.tsx:24]: This config should be applicable for every activity item. But but not for the whole activity unless it is a part of an activity flow

File: src/widgets/Header/lib/account-dropdown-options.constant.ts [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+
+    if (tokens?.accessToken) {
+      logoutMutation({ accessToken: tokens.accessToken })
+    }
+
+    clearUser()
+    clearActivityInProgressState()
+    secureTokensStorage.clearTokens()
+    navigate(ROUTES.login.path)
+  }
```
Comment (BamMironov) [inline src/widgets/Header/lib/account-dropdown-options.constant.ts:25]: This should be in a scope of the logout feature

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/widgets/ActivityDetails/model/hooks/useActivityInProgress.ts (hunk 3 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useActivityInProgress.ts:39]: Done

File: src/widgets/ActivityDetails/ui/ActivityItemList.tsx (hunk 4 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/ui/ActivityItemList.tsx:24]: This config is created on the top level and provided to children. In children  components, config can be changed regarding item settings.

File: src/widgets/Header/lib/account-dropdown-options.constant.ts (hunk 5 above)
Comment (moiskillnadne) [reply src/widgets/Header/lib/account-dropdown-options.constant.ts:25]: Done

File: src/entities/event/model/operations/ScheduledDateCalculator.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/entities/event/model/operations/ScheduledDateCalculator.ts]: Yes, it was a bug, fixed

File: src/entities/activity/model/activity.slice.ts [hunk 6]
```diff
@@ -1,58 +1,27 @@
...
+
+    upsertActivityById: (state, action: PayloadAction<ProgressPayloadState>) => {
+      const activityIndex = state.findIndex(el => el.activityId === action.payload.activityId)
+
+      if (activityIndex === -1) {
+        state.push(action.payload)
+        return
+      }
+
+      state[activityIndex] = action.payload
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/activity/model/activity.slice.ts]: Done

File: src/widgets/ActivityDetails/ui/ActivityItemList.tsx (hunk 4 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/ui/ActivityItemList.tsx:24]: I am working on it in another branch. This will be fixed in the next MR

**Review by BamMironov — APPROVED** [review]

**Pull request merged**